spec/index.bs

```diff
@@ -2028,13 +2028,48 @@ dictionary GPUProgrammableStageDescriptor {
 };
 </script>
 
+<script type=idl>
+enum GPUCompilationMessageType {
+    'error',
+    'warning',
+    'info'
+};
+enum GPUProgrammableStageType {
+    'vertex',
+    'fragment',
+    'compute'
+};
+
+[Serializable]
+interface GPUCompilationMessage {
+    GPUCompilationMessageType type;
```

Comment:
I’m pretty surprised that all these fields are present. Do we expect applications to use these fields at runtime to patch up their shaders dynamically?

Reply:
This looks like a structured representation of a standard error message to me. How would you see them being used for patching?